temba/channels/tests.py

```diff
@@ -4945,6 +4945,220 @@ def test_send(self):
+
+        # create an outgoing message instead
+        contact = msg2.contact
+        Msg.all_messages.all().delete()
+
+        contact.send("outgoing message", self.admin)
+        sms = Msg.all_messages.get()
+
+        # now update the status via a callback
+        twiml_api_url = reverse('handlers.twiml_api_handler', args=[self.channel.uuid]) + "?action=callback&id=%d" % sms.id
+        post_data['SmsStatus'] = 'sent'
+
+        signature = validator.compute_signature('https://' + settings.TEMBA_HOST + '%s' % twiml_api_url, post_data)
+        response = self.client.post(twiml_api_url, post_data, **{'HTTP_X_TWILIO_SIGNATURE': signature})
+
+        self.assertEquals(200, response.status_code)
+
+        sms = Msg.all_messages.get()
+        self.assertEquals(SENT, sms.status)
+
+        # try it with a failed SMS
+        Msg.all_messages.all().delete()
+        contact.send("outgoing message", self.admin)
+        sms = Msg.all_messages.get()
+
+        # now update the status via a callback (also test old api/v1 URL)
+        twiml_api_url = reverse('handlers.twiml_api_handler', args=[self.channel.uuid]) + "?action=callback&id=%d" % sms.id
+        post_data['SmsStatus'] = 'failed'
+
+        signature = validator.compute_signature('https://' + settings.TEMBA_HOST + '%s' % twiml_api_url, post_data)
+        response = self.client.post(twiml_api_url, post_data, **{'HTTP_X_TWILIO_SIGNATURE': signature})
+
+        self.assertEquals(200, response.status_code)
+        sms = Msg.all_messages.get()
+        self.assertEquals(FAILED, sms.status)
+
+    def test_send(self):
+
+        joe = self.create_contact("Joe", "+250788383383")
+        bcast = joe.send("Test message", self.admin, trigger_send=False)
+
+        # our outgoing sms
+        sms = bcast.get_messages()[0]
+
+        try:
+            settings.SEND_MESSAGES = True
+
+            with patch('twilio.rest.resources.Messages.create') as mock:
+                mock.return_value = "Sent"
+
+                # manually send it off
+                Channel.send_message(dict_to_struct('MsgStruct', sms.as_task_json()))
+
+                # check the status of the message is now sent
+                msg = bcast.get_messages()[0]
+                self.assertEquals(WIRED, msg.status)
+                self.assertTrue(msg.sent_on)
+
+                self.clear_cache()
+
+            with patch('twilio.rest.resources.Messages.create') as mock:
+                mock.side_effect = Exception("Failed to send message")
+
+                # manually send it off
+                Channel.send_message(dict_to_struct('MsgStruct', sms.as_task_json()))
+
+                # message should be marked as an error
+                msg = bcast.get_messages()[0]
+                self.assertEquals(ERRORED, msg.status)
+                self.assertEquals(1, msg.error_count)
+                self.assertTrue(msg.next_attempt)
+
+            # check that our channel log works as well
```

Comment:
No need for this or the tests below, just repeating testing of channel log which is already covered.